game/tournament.py

```diff
@@ -0,0 +1,51 @@
+import itertools
+from copy import copy
+from match import Match
+from player import Player
+from gameboard import GameBoard
+
+
+class Tournament(object):
+    def __init__(self, strategies):
+        self._pairs_of_strategies = itertools.permutations(strategies, 2)
+        self._number_of_matches = len([1 for i in self._pairs_of_strategies])
+        self._pairs_of_strategies = itertools.permutations(strategies, 2)
```

Comment:
I was resetting the iterator, it was needed after iterating through it on line 10, just solved it by creating a list at the beginning and just use that for the rest of the code